### PR TITLE
UICAL-175: Long exception periods display wrong days and subsequent e…

### PR DESCRIPTION
**Pull request by pavelspichonak**
## Purpose
Fix long exception periods display wrong days and subsequent exception periods display on the wrong days.

## Approach
Calendar can get dates which can have different timezone offset because of transition to winter time. So we have to get utc offset from each date instead of using constant `OFFSET_HOURS` which returns utc offset for today.

Also for performance improvement I've moved this logic from `startAccessor` and `endAccessor` to the place where events are created. It is necessary because `startAccessor` and `endAccessor` are called for cells which have events and which have no events.

## Refs
https://issues.folio.org/browse/UICAL-175

### CODE REVIEW

**Review by Stanislau-Snapok — APPROVED** [review]

**Review by Stanislau-Snapok — APPROVED** [review]

File: src/settings/utils/time.js [hunk 1]
```diff
@@ -1,2 +1,8 @@
+import moment from 'moment';
+
 // eslint-disable-next-line import/prefer-default-export
-export const OFFSET_HOURS = (new Date().getTimezoneOffset()) / 60;
+export const setTimezoneOffset = (date) => {
+  const momentDate = moment(date);
+  const utcOffset = -(momentDate.utcOffset() / 60);
+  return momentDate.add(utcOffset, 'hours');
```
Comment (pavelspichonak) [inline src/settings/utils/time.js:8]: add empty line before return

**Review by Dmitriy-Litvinenko — APPROVED** [review]

File: src/settings/utils/time.js (hunk 1 above)
Comment (pavelspichonak) [reply src/settings/utils/time.js:8]: done

**Comment by Dmitriy-Litvinenko** [comment]
Hello @zburke @mkuklis Could you please review?

**Review by zburke — APPROVED** [review]
Aha, this finally makes sense to me! If a period spans the DST transition, the offset on one side will line up with, e.g. 00:00:00 2021-01-01 to 23:59:59 2021-01-01, i.e. the same day, but on the other side of DST it will line up with, e.g. 01:00:00 2021-06-01 to 00:59:59 2021-06-02, i.e. a day ahead. I am so happy that I finally understand this bug!

**Comment by pavelspichonak** [comment]
> Aha, this finally makes sense to me! If a period spans the DST transition, the offset on one side will line up with, e.g. 00:00:00 2021-01-01 to 23:59:59 2021-01-01, i.e. the same day, but on the other side of DST it will line up with, e.g. 01:00:00 2021-06-01 to 00:59:59 2021-06-02, i.e. a day ahead. I am so happy that I finally understand this bug!

You are almost right.

We get start date for events from backend which look like this `2021-11-01T00:00:00.000+00:00`. It is always date without hours, minutes, seconds and milliseconds. When `moment` parses dates it converts dates to local time(using your local timezone offset). So if you live in Canada date `2021-11-01T00:00:00.000+00:00` becomes `Sun Oct 31 2021 20:00:00 GMT-0400 (GMT-04:00)`.

We need to add local timezone offset to dates for event to be put to the right cell in the Calendar. In case of Canada we need to add `4` hours for dates before `DST` transition and as far as I remember `5` for dates after `DST` transition because they have another timezone offset.

So in other words we need to get timezone offset from each date instead of using `const OFFSET_HOURS = (new Date().getTimezoneOffset()) / 60;` because this constant has the same timezone offset as dates before `DST` transition but not the same as dates after `DST` transition.

Or maybe you understood correctly and it is me who did not understand you :)

And thanks for your review!
